docs/contracts.rst

```diff
@@ -284,8 +284,23 @@ It will generate a function of the following form::
         b = data[arg1][arg2][arg3].b;
     }
 
-Note that the mapping in the struct is omitted because there
-is no good way to provide the key for the mapping.
+If you have a `public` state variable of array type, then it only returns individual elements to reduce the gas cost of returning an entire array. You can use parameters to specify which element to return, for example ``data[0]``. If you want to return an entire array, then you need to write a helper function, for example::
```

Comment:
This should probably be merged with the example above.

Reply:
@axic The

> If you have a `public` state variable of array type

section? I don't entirely understand the point, sorry.